h/services/search_index/_queue.py

```diff
@@ -28,78 +33,67 @@ def __init__(self, db, es, batch_indexer):
+            priority)
+        :param where: A list of SQLAlchemy BinaryExpression objects to limit
+            the annotations to be added
+        :param force: Whether to force reindexing of the annotation even if
+            it's already indexed
         :param schedule_in: A number of seconds from now to wait before making
             the job available for processing. The annotation won't be synced
             until at least `schedule_in` seconds from now
-        :type schedule_in: int
-
-        :param force: Whether to force reindexing of the annotation even if
-            it's already indexed
-        :type force: bool
         """
-        self.add_all([annotation_id], tag, schedule_in, force)
+        where_clause = and_(*where) if len(where) > 1 else where[0]
+        schedule_at = datetime.utcnow() + timedelta(seconds=schedule_in or 0)
+
+        query = Job.__table__.insert().from_select(
+            [Job.name, Job.tag, Job.priority, Job.kwargs, Job.scheduled_at],
+            select(
+                [
+                    text("'sync_annotation'"),
+                    text(repr(tag)),
+                    text(str(priority)),
+                    func.jsonb_build_object(
+                        "annotation_id", Annotation.id, "force", bool(force)
+                    ),
+                    text(f"'{schedule_at}'"),
+                ]
+            ).where(where_clause),
+        )
 
-    def add_all(self, annotation_ids, tag, schedule_in=None, force=False):
-        """
-        Queue a list of annotations to be synced to Elasticsearch.
+        self._db.execute(query)
+        mark_changed(self._db)
 
-        See Queue.add() for documentation of the params.
+    def add(self, annotation_id, tag, schedule_in=None, force=False):
         """
+        Queue an annotation to be synced to Elasticsearch.
 
-        # Jobs with a lower number for their priority get processed before jobs
-        # with a higher number. Make large batches of jobs added all at once
-        # get processed *after* small batches added a few at a time, so that
-        # large batches don't hold up small ones for a long time.
-        priority = len(annotation_ids)
-
-        self._db.add_all(
-            Job(
-                tag=tag,
-                name="sync_annotation",
-                scheduled_at=self._datetime_at(schedule_in),
-                priority=priority,
-                kwargs={
-                    "annotation_id": URLSafeUUID.url_safe_to_hex(annotation_id),
-                    "force": force,
-                },
-            )
-            for annotation_id in annotation_ids
-        )
+        See Queue.add_where() for documentation of the params.
+
+        :param annotation_id: The ID of the annotation to be queued, in the
+            application-level URL-safe format
+        """
+        where = [Annotation.id == annotation_id]
+        self.add_where(tag, self.Priority.SINGLE_ITEM, where, force, schedule_in)
```

Comment:
Passing through all the args and kwargs is annoying if we ever add more to the base one. We could `*kwargs` this, but it makes life harder for the caller.